celer/tests/test_lasso.py

```diff
@@ -218,6 +218,25 @@ def test_weights_lasso():
     np.testing.assert_raises(ValueError, clf1.fit, X=X, y=y)
 
 
+def test_infinite_weights():
+    np.random.seed(1)
+    n_samples, n_features = 50, 100
+    X, y = build_dataset(n_samples, n_features)
+
+    weights = np.ones(n_features)
+    nb_inf_index = n_features // 10
+    li_inf_index = np.random.randint(0, n_features+1, size=nb_inf_index)
+    weights[li_inf_index] = np.inf
+
+    alpha_max = norm(X.T @ y / weights, ord=np.inf) / n_samples
+
+    reg = Lasso(alpha=alpha_max / 100., weights=weights)
+    reg.fit(X, y)
+
+    # coef with inf weight should be set to 0
```

Comment:
can you assert that the solver has converged checking that the duality gap is smaller than the provided tol times 0.5 * norm(y) ** 2?

same for the other 2 estimators

Reply:
Yes, sure!